bitnami/jupyterhub/templates/proxy/tls-secret.yaml

```
@@ -0,0 +1,44 @@
+{{- if .Values.proxy.ingress.enabled }}
+{{- if .Values.proxy.ingress.secrets }}
+{{- range .Values.proxy.ingress.secrets }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .name }}
+  namespace: {{ $.Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    {{- if $.Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" $.Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+    app.kubernetes.io/component: proxy
+  {{- if $.Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: kubernetes.io/tls
+data:
+  tls.crt: {{ .certificate | b64enc }}
+  tls.key: {{ .key | b64enc }}
+---
+{{- end }}
+{{- else if and .Values.proxy.ingress.tls (not .Values.proxy.ingress.certManager) }}
```

Both using custom secrets and auto-generating one for the default hostname should be possible:
```suggestion
{{- end }}
{{- if and .Values.proxy.ingress.tls (not .Values.proxy.ingress.certManager) }}
```